apps/sasquatch/src/projectDependency/java/com/microsoft/appcenter/sasquatch/activities/MSALoginActivity.java

```diff
@@ -0,0 +1,396 @@
+
+/**
+ * TODO during release, delete jCenter version of this file and move this one to main source folder.
+ */
+public class MSALoginActivity extends AppCompatActivity {
+
+    private static final String URL_PREFIX = "https://login.live.com/oauth20_";
+
+    private static final String REDIRECT_URL = URL_PREFIX + "desktop.srf";
+
+    private static final String AUTHORIZE_URL = URL_PREFIX + "authorize.srf?";
+
+    private static final String TOKEN_URL = URL_PREFIX + "token.srf";
+
+    private static final String SIGN_OUT_URL = URL_PREFIX + "logout.srf?";
+
+    private static final String CLIENT_ID = "06181c2a-2403-437f-a490-9bcb06f85281";
+
+    private static final String[] SCOPES_COMPACT = {"service::events.data.microsoft.com::MBI_SSL"};
+
+    private static final String[] SCOPES_DELEGATE = {
+            "wl.offline_access",
+            "AsimovRome.Telemetry",
+    };
+
+    private static final String USER_ID = "user_id";
+
+    private static final String SCOPE = "scope";
+
+    private static final String REFRESH_TOKEN = "refresh_token";
+
+    private static final String CLIENT_ID_PARAM = "&client_id=" + CLIENT_ID;
+
+    private static final String REDIRECT_URI_PARAM;
+
+    static {
+        try {
+            REDIRECT_URI_PARAM = "redirect_uri=" + URLEncoder.encode(REDIRECT_URL, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * HTTP Client to get tokens.
+     */
+    private static HttpClient sHttpClient;
+
+    /**
+     * Refresh token if logged in.
+     */
+    private String mRefreshToken;
+
+    /**
+     * Refresh token scope.
+     */
+    private String mRefreshTokenScope;
+
+    /**
+     * Web view.
+     */
+    private WebView mWebView;
+
+    /**
+     * Authentication provider type.
+     */
+    private AuthenticationProvider.Type mAuthType;
+
+    @SuppressLint("SetJavaScriptEnabled")
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+
+        /* Init UI. */
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.msa_login);
+
+        Serializable rawAuthType = getIntent().getSerializableExtra(AuthenticationProvider.Type.class.getName());
+        mAuthType = (AuthenticationProvider.Type) rawAuthType;
+
+        /* Init API client only once. */
+        if (sHttpClient == null) {
+            sHttpClient = createHttpClient(this);
+        }
+
+        /* Configure web view. */
+        mWebView = findViewById(R.id.web_view);
+        mWebView.setWebChromeClient(new WebChromeClient());
+        mWebView.getSettings().setJavaScriptEnabled(true);
+
+        /* Show prompt or message that there will be no prompt to sign in. */
+        String cookie = CookieManager.getInstance().getCookie(AUTHORIZE_URL);
+        boolean compact = mAuthType == AuthenticationProvider.Type.MSA_COMPACT;
+        if (compact && cookie != null && cookie.contains("MSPPre")) {
```

Comment:
Move to const

Reply:
Thanks for the feedback but we can clean it up in another task. This is just copied from another place and we don't change any lines of code here.